src/indexer/services/evmIndexer/evmEventParser.ts

```diff
@@ -0,0 +1,193 @@
+  blockUnixTimestamp: number,
+  resourceMap: Map<string, Resource>,
+): DecodedDepositLog {
+  const resourceType = resourceMap.get(decodedLog.args.resourceID as string)?.type || ""
+  const resourceDecimals = resourceMap.get(decodedLog.args.resourceID as string)?.decimals || 18
+
+  const arrayifyData = getBytes(decodedLog.args.data as BytesLike)
+  const filtered = arrayifyData.filter((_, idx) => idx + 1 > 65)
+  const hexAddress = hexlify(filtered)
+  const destDomainID = decodedLog.args.destinationDomainID as number
+  return {
+    blockNumber: log.blockNumber,
+    depositNonce: Number(decodedLog.args.depositNonce),
+    toDomainId: destDomainID.toString(),
+    sender: txReceipt.from,
+    destination: hexAddress,
+    fromDomainId: domain.id.toString(),
+    resourceID: decodedLog.args.resourceID as string,
+    txHash: log.transactionHash,
+    timestamp: blockUnixTimestamp,
+    depositData: decodedLog.args.data as string,
+    handlerResponse: decodedLog.args.handlerResponse as string,
+    transferType: resourceType,
+    amount: decodeAmountsOrTokenId(decodedLog.args.data as string, resourceDecimals, resourceType),
+  }
+}
+
+export function parseProposalExecution(
+  log: Log,
+  decodedLog: LogDescription,
+  txReceipt: TransactionReceipt,
+  blockUnixTimestamp: number,
+  resourceMap: Map<string, Resource>,
+): DecodedProposalExecutionLog {
+  const resourceType = resourceMap.get(decodedLog.args.resourceID as string)?.type || ""
+  const originDomainID = decodedLog.args.originDomainID as number
+  return {
+    blockNumber: log.blockNumber,
+    from: txReceipt.from,
+    depositNonce: Number(decodedLog.args.depositNonce as string),
+    txHash: log.transactionHash,
+    timestamp: blockUnixTimestamp,
+    fromDomainId: originDomainID.toString(),
+    transferType: resourceType,
+    resourceID: decodedLog.args.resourceID as string,
+  }
+}
+
+export async function parseFeeCollected(
+  decodedLog: LogDescription,
+  provider: Provider,
+  nativeTokenSymbol: string,
+  log: Log,
+): Promise<DecodedFeeCollectedLog> {
+  const ercToken = getERC20Contract(provider, decodedLog.args.tokenAddress as string)
+
+  return {
+    amount: BigNumber.from(decodedLog.args.fee as number).toString(),
+    tokenSymbol: (decodedLog.args.tokenAddress as string) == nativeTokenAddress ? nativeTokenSymbol : (await ercToken?.symbol()) || "",
+    tokenAddress: decodedLog.args.tokenAddress as string,
+    txHash: log.transactionHash,
+  }
+}
+
+export function parseFailedHandlerExecution(log: Log, decodedLog: LogDescription): DecodedFailedHandlerExecution {
+  const originDomainID = decodedLog.args.originDomainID as number
+  return {
+    domainId: originDomainID.toString(),
+    depositNonce: Number(decodedLog.args.depositNonce as string),
+    txHash: log.transactionHash,
+    blockNumber: log.blockNumber,
+  }
+}
+
+function decodeAmountsOrTokenId(data: string, decimals: number, resourceType: string): string {
+  if (resourceType === "fungible") {
```

Comment:
Make this switch and throw on default case if nothing matches

Reply:
done